tests/utils/queryApiGateway.test.tsx

```diff
@@ -0,0 +1,150 @@
+import queryApiGateway, { buildApiUrl, extractJSON, Claim, QueryParams } from '../../utils/queryApiGateway'
+import mockEnv from 'mocked-env'
+import fs from 'fs'
+import fetch from 'node-fetch'
+
+// Mock some modules
+jest.mock('fs')
+jest.mock('node-fetch')
+// const { Response } = jest.requireActual('node-fetch')
+
+// Shared test constants
+const goodUrl = 'http://nowhere.com'
+
+/**
+ * Begin tests
+ */
+
+// Test queryApiGateway
+describe('Querying the API Gateway', () => {
+  const goodResponse = { ClaimType: 'PUA' }
+  const goodRequest = {
+    headers: {
+      id: '12345',
+    },
+  }
+
+  beforeEach(() => {
+    // Mock the fetch response
+    // fetch.mockReturnValue(Promise.resolve(new Response(JSON.stringify(goodResponse))))
+    // Mock fs.readFileSync()
+    fs.readFileSync = jest.fn().mockResolvedValue('mock file data')
+  })
+
+  afterEach(() => {
+    // Clear mocks after each test
+    jest.clearAllMocks()
+  })
+
+  // Test mock fetch is working correctly
+  it('has a correctly mocked fetch', async () => {
+    const resp: Response = await fetch()
+    const body: string = await resp.text()
+    const jsonData: Claim = extractJSON(body)
+    expect(jsonData.ClaimType).toBe('PUA')
+    expect(fetch).toHaveBeenCalledTimes(1)
+  })
+
+  // Test that queryApiGateway works correctly
+  it('returns the expected value', async () => {
+    // Mock process.env
+    const restore = mockEnv({
+      API_URL: goodUrl,
+    })
+
+    const data = await queryApiGateway(goodRequest)
+    expect(data).toStrictEqual(goodResponse)
+    expect(fetch).toHaveBeenCalledTimes(1)
+
+    // Restore env vars
+    restore()
+  })
+
+  // Test that queryApiGateway handles Errors
+  it('handles errors thrown by fetch', async () => {
+    // Mock a network error
+    // fetch.mockRejectedValueOnce(new Error('network error'))
```

Comment:
This line should be uncommented, but needs to pass linting before it can be commited